src/astro/databases/google/bigquery.py

```diff
@@ -281,11 +282,6 @@ def load_s3_file_to_bigquery(
 
         project_id = self.get_project_id(target_table)
 
-        if if_exists == "replace":
```

Comment:
Just FYI - In this function, it may be fine, but we should pass the `if_exists` param, since not all the table creation is done via the create_table() function, we are also using bigquery's autodetection in GCS to Bigquery path.

Reply:
Should we allow the user to define the schema using these three methods?
1. explicitly defined columns
2. auto-detection using dataframe
3. auto-detection using native implementation

Perhaps we could focus on the first two methods - I noticed that most of the Snowflake implementation complexity was actually trying to use Snowflake to infer the schema.

If we focus on the first two methods, this gives an evident boundary between creating the table x loading data to an existing table and reduces possible issues. Time can tell us if users care about native schema detection!

Reply:
I would be in favor of including (3) too where the support is straightforward. For Bigquery, _**I think**_ native support to detect schema might be very mature

Reply:
Spoke to @tatiana and since we are using autodetect for bigquery only and no other databases support it across all file types. It would be better to take a generic approach to detect schema, which works for all db and file types. We won't be using auto schema detection from native implementation(approach 3), only approach 1  and 2 will be used.

Reply:
@kaxil @utkarsharma2 I agree we could also support (3) for BQ. This could be done without much overhead - but it may be worth waiting and seeing if this will be a requirement for users. If you think this has high priority, I can do it in a separate PR, after I finished the Snowflake load part using `COPY INTO` - which becomes much simpler after this PR is merged 🤞